src/bin/scheduler/setup/directories.rs

```diff
@@ -368,6 +370,66 @@ fn clean_up_results_directory(
     Ok(results_directory_lock.release()?)
 }
 
+struct OwnershipSetter {
+    #[cfg(unix)]
+    user_id: u32,
+    #[cfg(unix)]
+    group_id: u32,
+    #[cfg(windows)]
+    user: String,
+}
+
+impl OwnershipSetter {
+    pub fn make_for_current_user() -> AnyhowResult<Self> {
+        #[cfg(unix)]
+        {
+            Ok(Self {
+                user_id: unsafe { libc::getuid() },
+                group_id: unsafe { libc::getgid() },
+            })
+        }
+        #[cfg(windows)]
+        {
+            Ok(Self {
+                user: whoami::username(),
```

Comment:
I think this output needs to be sanitized? `whoami` crate can return `Unknown` in some circumstances? Do we really want to change the permissions of the directories to that of the user `Unknown`? I feel that the whoami crate is creating more problems then is solving tbh.